### PR TITLE
Varlamore Patch

### PR DESCRIPTION
**Pull request by Macweese**
This PR is intended on adding most Varlamore additions such as world map additions, agility shortcuts, regions for Discord rich presence, farming tracking and item stats.

I hope this is not too many additions at once. Please let me know if I missed something or amendments are preferred, e.g. squashing commits.

Leaving a note:
The dungeons inside Cam Torum have the same region IDs as the city, presumably they are just on different planes. The current implementation for Discord regions does not allow duplicate region IDs, so the dungeons were omitted.

<details>
  <summary>Omitted regions</summary>
  
  ```java
	DUNGEON_NEYPOTZLI_ANCIENT_PRISON("Neypotzli Ancient Prison", DiscordAreaType.DUNGEONS, 5525),
	DUNGEON_NEYPOTZLI_ANCIENT_SHRINE("Neypotzli Ancient Shrine", DiscordAreaType.DUNGEONS, 6037),
	DUNGEON_NEYPOTZLI_ANTECHAMBER("Neypotzli Antechamber", DiscordAreaType.DUNGEONS, 5781),
	DUNGEON_NEYPOTZLI_EARTHBOUND_CAVERN("Neypotzli Earthbound Cavern", DiscordAreaType.DUNGEONS, 5527),
	DUNGEON_NEYPOTZLI_STREAMBOUND_CAVERN("Neypotzli Streambound Cavern", DiscordAreaType.DUNGEONS, 6039),
	DUNGEON_NEYPOTZLI_THE_BLOOD_MOON("Neypotzli The Blood Moon", DiscordAreaType.DUNGEONS, 5526),
	DUNGEON_NEYPOTZLI_THE_BLUE_MOON("Neypotzli The Blue Moon", DiscordAreaType.DUNGEONS, 5783),
	DUNGEON_NEYPOTZLI_THE_ECLIPSE_MOON("Neypotzli The Eclipse Moon", DiscordAreaType.DUNGEONS, 6038),
  ```
  
</details>

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/worldmap/FairyRingLocation.java [hunk 1]
```diff
@@ -35,6 +35,7 @@ enum FairyRingLocation
 	AIR("AIR", new WorldPoint(2699, 3249, 0)),
 	// AJQ - Exists in game but not on World Map
 	AJR("AJR", new WorldPoint(2779, 3615, 0)),
+	AJP("AJP", new WorldPoint(1648, 3011, 0)),
```
Comment (testing-ongithub) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldmap/FairyRingLocation.java:38]: Could you move this a couple lines up so this list stays alphabetized? 😃

**Comment by Nightfirecat** [comment]
I've cherry-picked (with fixes and adaptations where needed) many of these commits, though there are still a few I would like to review--mostly because I haven't had time to actually interact with the new content and unlock access to all the new areas so can't confirm with certainty every data point--from this PR, namely:

- [`c3b5778` (#17588)](https://github.com/runelite/runelite/pull/17588/commits/c3b5778b0496f41a2700d567386a1a3459802476) (discord regions)
- [`33842fb` (#17588)](https://github.com/runelite/runelite/pull/17588/commits/33842fbbb203cee05be0c8b9571936e55bd0dbb8) (moth item stats)

**Comment by Nightfirecat** [comment]
> The dungeons inside Cam Torum have the same region IDs as the city, presumably they are just on different planes. The current implementation for Discord regions does not allow duplicate region IDs, so the dungeons were omitted.

This does not seem to be true at all. The room containing the monolith appears to be region 22, 150 (5782), and the others similarly are not overlapping the city. The world map location dev tool is not always accurate--particularly for dungeon locations--so the *Location* dev tool, or shell, should be used to verify them instead.

**Comment by Nightfirecat** [comment]
Discord areas have been added in 6b793fc0e5095c7f63eeeda4ae091d691e23c7b3, and the item stat changes are being reviewed separately in #17607, so there is nothing more to be reviewed from this PR. Thanks for the help with all these changes.

**Pull request closed without merging**